apps/condo/domains/marketplace/schema/MarketCategory.test.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * Generated by `createschema marketplace.MarketCategory 'name:Text; parentCategory:Relationship:MarketCategory:CASCADE;'`
+ */
+
+const { makeLoggedInAdminClient, makeClient, expectValuesOfCommonFields, expectToThrowGQLError, expectToThrowInternalError } = require('@open-condo/keystone/test.utils')
+const {
+    expectToThrowAuthenticationErrorToObj, expectToThrowAuthenticationErrorToObjects,
+    expectToThrowAccessDeniedErrorToObj,
+} = require('@open-condo/keystone/test.utils')
+
+
+const { MARKET_CATEGORY_ERRORS } = require('@condo/domains/marketplace/constants')
+const { MarketCategory, createTestMarketCategory, updateTestMarketCategory } = require('@condo/domains/marketplace/utils/testSchema')
+const { makeClientWithNewRegisteredAndLoggedInUser, makeClientWithSupportUser } = require('@condo/domains/user/utils/testSchema')
+
+describe('MarketCategory', () => {
+    let admin, support, anonymous, user, checkReadAccessCategoryObj
+    beforeAll(async () => {
+        admin = await makeLoggedInAdminClient()
+        support = await makeClientWithSupportUser()
+        user = await makeClientWithNewRegisteredAndLoggedInUser()
+        anonymous = await makeClient();
+        [checkReadAccessCategoryObj] = await createTestMarketCategory(admin)
+    })
+    describe('access tests', () => {
+        describe('admin', () => {
+            test('can create', async () => {
+                const [obj, attrs] = await createTestMarketCategory(admin)
+                expectValuesOfCommonFields(obj, attrs, admin)
+            })
+
+            test('can update', async () => {
+                const [objCreated] = await createTestMarketCategory(admin)
+
+                const [obj, attrs] = await updateTestMarketCategory(admin, objCreated.id)
+
+                expect(obj.dv).toEqual(1)
+                expect(obj.sender).toEqual(attrs.sender)
+                expect(obj.v).toEqual(2)
+                expect(obj.updatedBy).toEqual(expect.objectContaining({ id: admin.user.id }))
+            })
+
+            test('can\'t delete', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await MarketCategory.delete(admin, 'id')
+                })
+            })
+
+            test('can read', async () => {
+                const objs = await MarketCategory.getAll(admin, {}, { sortBy: ['updatedAt_DESC'] })
```

Comment:
Maybe it would be better to indicate the desired ID in the filter?
Someday there will be a lot of these objects and problems may begin...

Reply:
for tests a clean base is raised, it is unlikely that there will be 100 tests with the creation of categories in each